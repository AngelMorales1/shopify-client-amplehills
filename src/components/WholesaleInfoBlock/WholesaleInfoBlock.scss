@import 'settings', 'typography', 'src/styles/utils';

.WholesaleInfoBlock {
  flex-direction: column;
  padding: spacing(4) spacing(8);
  margin-top: spacing(40);

  @media ($medium-up) {
    flex-direction: row;
    padding: spacing(16);
  }
}
