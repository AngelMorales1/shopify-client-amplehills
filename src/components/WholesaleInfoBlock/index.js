import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { Image, Button } from 'components/base';
import marked from 'marked';

import styles from './WholesaleInfoBlock.scss';

const WholesaleInfoBlock = ({ image, title, description }) => {
  return (
    <div
      className={cx(
        styles['WholesaleInfoBlock'],
        'flex items-center justify-between bg-lavender-blush mx3'
      )}
    >
      <div className="flex flex-row justify-center items-center col-12 md-col-7 py3">
        <Image className="col-12 md-col-8" src={image} />
      </div>
      <div className="flex flex-column col-12 md-col-5 my4">
        <p className="description-title bold mb2 text-container-width">
          {title}
        </p>
        <div
          dangerouslySetInnerHTML={{
            __html: marked(description)
          }}
          className="markdown-block text-container-width col-12 md-col-9"
        />
        <Button
          className="inline-flex mt3"
          to="/wholesale"
          label="Wholesale Info"
          color="peach"
        />
      </div>
    </div>
  );
};

WholesaleInfoBlock.propTypes = {
  image: PropTypes.string,
  title: PropTypes.string,
  description: PropTypes.string
};

WholesaleInfoBlock.defaultProps = {
  image: '',
  title: '',
  description: ''
};

export default WholesaleInfoBlock;
