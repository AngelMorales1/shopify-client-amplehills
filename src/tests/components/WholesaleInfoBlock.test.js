import React from 'react';
import { shallow } from 'enzyme';

import WholesaleInfoBlock from 'components/WholesaleInfoBlock';

it('renders without data', () => {
  const component = shallow(<WholesaleInfoBlock />);

  expect(component).toMatchSnapshot();
});

it('renders with data', () => {
  const component = shallow(
    <WholesaleInfoBlock image="foo" title="foo" description="foo" />
  );

  expect(component).toMatchSnapshot();
});
