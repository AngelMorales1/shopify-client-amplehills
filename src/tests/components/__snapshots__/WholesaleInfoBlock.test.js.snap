// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WholesaleInfoBlock
    description="foo"
    image="foo"
    title="foo"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="flex flex-row justify-center items-center col-12 md-col-7 py3"
>
          <Image
                    alt=""
                    bg={false}
                    className="col-12 md-col-8"
                    onImgLoad={[Function]}
                    src="foo"
                    style={Object {}}
          />
</div>,
        <div
          className="flex flex-column col-12 md-col-5 my4"
>
          <p
                    className="description-title bold mb2 text-container-width"
          >
                    foo
          </p>
          <div
                    className="markdown-block text-container-width col-12 md-col-9"
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": "<p>foo</p>
                              ",
                                      }
                    }
          />
          <Button
                    childrenWrapperClassName=""
                    className="inline-flex"
                    color="peach"
                    disabled={false}
                    fullWidth={false}
                    hover="shadow"
                    label="Wholesale Info"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    openInCurrentTab={false}
                    shadow={false}
                    to="/wholesale"
                    type="button"
                    variant="primary-responsive"
          />
</div>,
      ],
      "className": "flex items-center justify-between bg-lavender-blush mx3",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Image
            alt=""
            bg={false}
            className="col-12 md-col-8"
            onImgLoad={[Function]}
            src="foo"
            style={Object {}}
/>,
          "className": "flex flex-row justify-center items-center col-12 md-col-7 py3",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alt": "",
            "bg": false,
            "children": null,
            "className": "col-12 md-col-8",
            "onImgLoad": [Function],
            "src": "foo",
            "style": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="description-title bold mb2 text-container-width"
>
              foo
</p>,
            <div
              className="markdown-block text-container-width col-12 md-col-9"
              dangerouslySetInnerHTML={
                            Object {
                                          "__html": "<p>foo</p>
                            ",
                                        }
              }
/>,
            <Button
              childrenWrapperClassName=""
              className="inline-flex"
              color="peach"
              disabled={false}
              fullWidth={false}
              hover="shadow"
              label="Wholesale Info"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              openInCurrentTab={false}
              shadow={false}
              to="/wholesale"
              type="button"
              variant="primary-responsive"
/>,
          ],
          "className": "flex flex-column col-12 md-col-5 my4",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "foo",
              "className": "description-title bold mb2 text-container-width",
            },
            "ref": null,
            "rendered": "foo",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "markdown-block text-container-width col-12 md-col-9",
              "dangerouslySetInnerHTML": Object {
                "__html": "<p>foo</p>
",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": null,
              "childrenWrapperClassName": "",
              "className": "inline-flex",
              "color": "peach",
              "disabled": false,
              "fullWidth": false,
              "hover": "shadow",
              "label": "Wholesale Info",
              "onClick": [Function],
              "onMouseEnter": [Function],
              "onMouseLeave": [Function],
              "openInCurrentTab": false,
              "shadow": false,
              "to": "/wholesale",
              "type": "button",
              "variant": "primary-responsive",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="flex flex-row justify-center items-center col-12 md-col-7 py3"
>
            <Image
                        alt=""
                        bg={false}
                        className="col-12 md-col-8"
                        onImgLoad={[Function]}
                        src="foo"
                        style={Object {}}
            />
</div>,
          <div
            className="flex flex-column col-12 md-col-5 my4"
>
            <p
                        className="description-title bold mb2 text-container-width"
            >
                        foo
            </p>
            <div
                        className="markdown-block text-container-width col-12 md-col-9"
                        dangerouslySetInnerHTML={
                                    Object {
                                                "__html": "<p>foo</p>
                                    ",
                                              }
                        }
            />
            <Button
                        childrenWrapperClassName=""
                        className="inline-flex"
                        color="peach"
                        disabled={false}
                        fullWidth={false}
                        hover="shadow"
                        label="Wholesale Info"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        openInCurrentTab={false}
                        shadow={false}
                        to="/wholesale"
                        type="button"
                        variant="primary-responsive"
            />
</div>,
        ],
        "className": "flex items-center justify-between bg-lavender-blush mx3",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Image
              alt=""
              bg={false}
              className="col-12 md-col-8"
              onImgLoad={[Function]}
              src="foo"
              style={Object {}}
/>,
            "className": "flex flex-row justify-center items-center col-12 md-col-7 py3",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "",
              "bg": false,
              "children": null,
              "className": "col-12 md-col-8",
              "onImgLoad": [Function],
              "src": "foo",
              "style": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="description-title bold mb2 text-container-width"
>
                foo
</p>,
              <div
                className="markdown-block text-container-width col-12 md-col-9"
                dangerouslySetInnerHTML={
                                Object {
                                                "__html": "<p>foo</p>
                                ",
                                              }
                }
/>,
              <Button
                childrenWrapperClassName=""
                className="inline-flex"
                color="peach"
                disabled={false}
                fullWidth={false}
                hover="shadow"
                label="Wholesale Info"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                openInCurrentTab={false}
                shadow={false}
                to="/wholesale"
                type="button"
                variant="primary-responsive"
/>,
            ],
            "className": "flex flex-column col-12 md-col-5 my4",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "foo",
                "className": "description-title bold mb2 text-container-width",
              },
              "ref": null,
              "rendered": "foo",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "markdown-block text-container-width col-12 md-col-9",
                "dangerouslySetInnerHTML": Object {
                  "__html": "<p>foo</p>
",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": null,
                "childrenWrapperClassName": "",
                "className": "inline-flex",
                "color": "peach",
                "disabled": false,
                "fullWidth": false,
                "hover": "shadow",
                "label": "Wholesale Info",
                "onClick": [Function],
                "onMouseEnter": [Function],
                "onMouseLeave": [Function],
                "openInCurrentTab": false,
                "shadow": false,
                "to": "/wholesale",
                "type": "button",
                "variant": "primary-responsive",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WholesaleInfoBlock
    description=""
    image=""
    title=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="flex flex-row justify-center items-center col-12 md-col-7 py3"
>
          <Image
                    alt=""
                    bg={false}
                    className="col-12 md-col-8"
                    onImgLoad={[Function]}
                    src=""
                    style={Object {}}
          />
</div>,
        <div
          className="flex flex-column col-12 md-col-5 my4"
>
          <p
                    className="description-title bold mb2 text-container-width"
          />
          <div
                    className="markdown-block text-container-width col-12 md-col-9"
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": "",
                                      }
                    }
          />
          <Button
                    childrenWrapperClassName=""
                    className="inline-flex"
                    color="peach"
                    disabled={false}
                    fullWidth={false}
                    hover="shadow"
                    label="Wholesale Info"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    openInCurrentTab={false}
                    shadow={false}
                    to="/wholesale"
                    type="button"
                    variant="primary-responsive"
          />
</div>,
      ],
      "className": "flex items-center justify-between bg-lavender-blush mx3",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Image
            alt=""
            bg={false}
            className="col-12 md-col-8"
            onImgLoad={[Function]}
            src=""
            style={Object {}}
/>,
          "className": "flex flex-row justify-center items-center col-12 md-col-7 py3",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alt": "",
            "bg": false,
            "children": null,
            "className": "col-12 md-col-8",
            "onImgLoad": [Function],
            "src": "",
            "style": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="description-title bold mb2 text-container-width"
/>,
            <div
              className="markdown-block text-container-width col-12 md-col-9"
              dangerouslySetInnerHTML={
                            Object {
                                          "__html": "",
                                        }
              }
/>,
            <Button
              childrenWrapperClassName=""
              className="inline-flex"
              color="peach"
              disabled={false}
              fullWidth={false}
              hover="shadow"
              label="Wholesale Info"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              openInCurrentTab={false}
              shadow={false}
              to="/wholesale"
              type="button"
              variant="primary-responsive"
/>,
          ],
          "className": "flex flex-column col-12 md-col-5 my4",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "description-title bold mb2 text-container-width",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "markdown-block text-container-width col-12 md-col-9",
              "dangerouslySetInnerHTML": Object {
                "__html": "",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": null,
              "childrenWrapperClassName": "",
              "className": "inline-flex",
              "color": "peach",
              "disabled": false,
              "fullWidth": false,
              "hover": "shadow",
              "label": "Wholesale Info",
              "onClick": [Function],
              "onMouseEnter": [Function],
              "onMouseLeave": [Function],
              "openInCurrentTab": false,
              "shadow": false,
              "to": "/wholesale",
              "type": "button",
              "variant": "primary-responsive",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="flex flex-row justify-center items-center col-12 md-col-7 py3"
>
            <Image
                        alt=""
                        bg={false}
                        className="col-12 md-col-8"
                        onImgLoad={[Function]}
                        src=""
                        style={Object {}}
            />
</div>,
          <div
            className="flex flex-column col-12 md-col-5 my4"
>
            <p
                        className="description-title bold mb2 text-container-width"
            />
            <div
                        className="markdown-block text-container-width col-12 md-col-9"
                        dangerouslySetInnerHTML={
                                    Object {
                                                "__html": "",
                                              }
                        }
            />
            <Button
                        childrenWrapperClassName=""
                        className="inline-flex"
                        color="peach"
                        disabled={false}
                        fullWidth={false}
                        hover="shadow"
                        label="Wholesale Info"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        openInCurrentTab={false}
                        shadow={false}
                        to="/wholesale"
                        type="button"
                        variant="primary-responsive"
            />
</div>,
        ],
        "className": "flex items-center justify-between bg-lavender-blush mx3",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Image
              alt=""
              bg={false}
              className="col-12 md-col-8"
              onImgLoad={[Function]}
              src=""
              style={Object {}}
/>,
            "className": "flex flex-row justify-center items-center col-12 md-col-7 py3",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "",
              "bg": false,
              "children": null,
              "className": "col-12 md-col-8",
              "onImgLoad": [Function],
              "src": "",
              "style": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="description-title bold mb2 text-container-width"
/>,
              <div
                className="markdown-block text-container-width col-12 md-col-9"
                dangerouslySetInnerHTML={
                                Object {
                                                "__html": "",
                                              }
                }
/>,
              <Button
                childrenWrapperClassName=""
                className="inline-flex"
                color="peach"
                disabled={false}
                fullWidth={false}
                hover="shadow"
                label="Wholesale Info"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                openInCurrentTab={false}
                shadow={false}
                to="/wholesale"
                type="button"
                variant="primary-responsive"
/>,
            ],
            "className": "flex flex-column col-12 md-col-5 my4",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "description-title bold mb2 text-container-width",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "markdown-block text-container-width col-12 md-col-9",
                "dangerouslySetInnerHTML": Object {
                  "__html": "",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": null,
                "childrenWrapperClassName": "",
                "className": "inline-flex",
                "color": "peach",
                "disabled": false,
                "fullWidth": false,
                "hover": "shadow",
                "label": "Wholesale Info",
                "onClick": [Function],
                "onMouseEnter": [Function],
                "onMouseLeave": [Function],
                "openInCurrentTab": false,
                "shadow": false,
                "to": "/wholesale",
                "type": "button",
                "variant": "primary-responsive",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
